login-workflow/src/new-architecure/components/workflow/WorkflowCardActions.tsx

```diff
@@ -0,0 +1,91 @@
+import { getWorkflowCardActionsUtilityClass, WorkflowCardActionsClassKey } from './Utility';
+import { unstable_composeClasses as composeClasses } from '@mui/base';
+
+const useUtilityClasses = (ownerState: WorkflowCardActionsProps): Record<WorkflowCardActionsClassKey, string> => {
+    const { classes } = ownerState;
+
+    const slots = {
+        root: ['root'],
+        nextButton: ['nextButton'],
+        previousButton: ['previousButton'],
+        stepper: ['stepper'],
+    };
+
+    return composeClasses(slots, getWorkflowCardActionsUtilityClass, classes);
+};
+
+export const WorkflowCardActions: React.FC<WorkflowCardActionsProps> = (props) => {
+    const {
+        canGoNext,
+        canGoPrevious,
+        showPrevious,
+        showNext,
+        previousLabel,
+        nextLabel,
+        onPrevious,
+        onNext,
+        currentStep,
+        totalSteps,
+        fullWidthButton,
+        sx,
+        ...cardActionsProps
+    } = props;
+    const defaultClasses = useUtilityClasses(props);
+
+    return (
+        <CardActions
+            sx={[{ justifyContent: 'flex-end', p: { sm: 2, md: 3 } }, ...(Array.isArray(sx) ? sx : [sx])]}
+            className={cx(defaultClasses.root)}
+            {...cardActionsProps}
+        >
+            <MobileStepper
+                variant={'dots'}
+                position={'static'}
+                steps={totalSteps}
+                activeStep={currentStep}
+                backButton={
+                    showPrevious ? (
+                        <Button
+                            variant="outlined"
+                            color="primary"
+                            disabled={
+                                canGoPrevious === false || (typeof canGoPrevious === 'function' && !canGoPrevious())
+                            }
+                            onClick={onPrevious}
+                            sx={[{ width: fullWidthButton ? '100%' : 100 }, ...(Array.isArray(sx) ? sx : [sx])]}
+                            className={cx(defaultClasses.previousButton)}
+                        >
+                            {previousLabel}
+                        </Button>
+                    ) : null
+                }
+                nextButton={
+                    showNext ? (
+                        <Button
+                            variant="contained"
+                            color="primary"
+                            disableElevation
+                            disabled={canGoNext === false || (typeof canGoNext === 'function' && !canGoNext())}
+                            onClick={onNext}
+                            sx={[{ width: fullWidthButton ? '100%' : 100 }, ...(Array.isArray(sx) ? sx : [sx])]}
+                            className={cx(defaultClasses.nextButton)}
+                        >
+                            {nextLabel}
+                        </Button>
+                    ) : null
+                }
+                sx={[
```

Comment:
I don't see anything in here handling the case when there is a single button (hiding the dots)...does this work as expected (I would've though we'd need some style for it)?